### PR TITLE
feat(card): Card actions support to custom content

### PR DESCRIPTION
**Pull request by cgirani**
**Summary**

- Adding support to card actions to array with custom content, so we can send custom elements to be used

**Change List (commits, features, bugs, etc)**

- extraActions support to array
- updates in MDX

**Acceptance Test (how to verify the PR)**

- Check that custom content can be sent in the extraActions prop following the pattern of array with objects

**Regression Test (how to make sure this PR doesn't break old functionality)**

- Check that none of the changes impacted the card header

**Things to look for during review**

- [ ] Make sure all references to `iot` or `bx` class prefix is using the prefix variable
- [ ] (React) All major areas have a `data-testid` attribute. New test ids should have test written to ensure they are not changed or removed.
- [ ] UI should be checked in RTL mode to ensure the proper handling of layout and text.
- [ ] All strings should be translatable.
- [ ] The code should pass a11y scans (The storybook a11y knob should show no violations). New components should have a11y test written.
- [ ] Unit test should be written and should have a coverage of 90% or higher in all areas.
- [ ] All components should be passing visual regression test. For new styles or components either a visual regression test should be written for all permutations or the base image updated.
- [ ] Changes or new components should either write new or update existing documentation.
- [ ] PR should link and close out an existing issue

### CODE REVIEW

**Comment by davidicus** [comment]
@dianatran18 I think you have to weight in on this since this changes the existing style guidance for this component. You can see an example [here](https://deploy-preview-3543--carbon-addons-iot-react.netlify.app/?path=/story/1-watson-iot-card-card--basic-with-custom-action)

**Comment by dianatran18** [comment]
Hey @cgirani do you have an example of the design that is requesting the custom actions?

**Review by davidicus — COMMENTED** [review]
I'm not quite sure why we do not have an extraActions in the prop type declarations or why eslint does not scream at you for it but we should probably add here to document. Otherwise, this looks good code wise. I have asked @dianatran18 to weigh in on the design as this changes the guidance. Also our translated string for extra actions label may not be adequate with multiples so may be something we need to address as well.

**Comment by dianatran18** [comment]
I think custom actions are fine to have if that's a need for the team, but there should be some structure around spacing/padding, order of actions, and number of actions if so. We also have the footer in the card component that can also be used to place additional actions, so it doesn't overcrowd the card header where the title also lives.

**Comment by cgirani** [comment]
@dianatran18 / @davidicus  this is what Luke had for us:
image

**Comment by cgirani** [comment]
@dianatran18 thanks for following up with Luke on that!
just a small question @davidicus, we would still need this PR right, at least since at this moment the Switcher is not supported in actions or we will add support only for the switcher?

**Review by herleraja — APPROVED** [review]
LGTM

**Comment by davidicus** [comment]
@cgirani yes. we will still need this PR. I am doing another review.

**Review by davidicus — CHANGES_REQUESTED** [review]
Hey @cgirani. This looks good. We just need to address the two comments about the prop type and the translated string. Then if we can update the story to show the new design using the footer for the text button.

File: packages/react/src/constants/CardPropTypes.js [hunk 1]
```diff
@@ -966,6 +966,7 @@ export const CardPropTypes = {
         hidden: PropTypes.bool,
       })
     ),
+    content: PropTypes.arrayOf(PropTypes.node),
```
Comment (davidicus) [inline packages/react/src/constants/CardPropTypes.js:969]: This should probably reflect the option of a single object which is the current prop signature?

Comment (cgirani) [reply packages/react/src/constants/CardPropTypes.js]: Yeah I changed it to be just the node... and also some other small updates

**Review by davidicus — APPROVED** [review]

**Pull request merged**